homeassistant/components/ssdp/__init__.py

```diff
@@ -85,6 +86,29 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+_SsdpServiceInfoBase = TypedDict(
+    "_SsdpServiceInfoBase",
+    {
+        "x-homeassistant-matching-domains": set,
+    },
+    total=True,
```

Comment:
`USN` is a mandatory field. It can serve as a `unique_id` if you do decide to have a common base-class for the discovery dict types.

Also, either ST or NT will always be present, and `discovery_info_from_headers_and_description` currently adds ST from NT, so we could make ST a mandatory field too.

And `UDN` (`ATTR_UPNP_UDN`) is always in the final SSDP + UPnP info.